remotespark/livyclientlib/livysession.py

```diff
@@ -106,11 +107,38 @@ def create_sql_context(self):
         else:
             raise FailedToCreateSqlContextException("Failed to create the SqlContext.\nError: '{}'".format(out))
 
+    def get_app_id(self):
+        if self._app_id is None:
+            session = self._http_client.get_session(self.id)
+            self._app_id = session['appId']
```

Comment:
I mean, we do have Livy versioning (a set of Livy versions we support and don't support), it's just not specified.

I would be comfortable with either 1) trying to do this in a backwards-compatible fashion by checking whether the record includes a session ID and 2) pinning the lowest supported version of Livy for publicpreview0.5 to whatever the current version is